### PR TITLE
Fix issue 18111: use fully qualified name for modules in unittest names

### PR DESCRIPTION
**Pull request by atilaneves**

### CODE REVIEW

File: src/dmd/dmodule.d
Comment (JinShil) [inline src/dmd/dmodule.d]: I don't see any reason for the extra newline.

**Review by JinShil — APPROVED** [review]

**Comment by wilzbach** [comment]
Shouldn't there be a test that ensures that the issue can't happen in the future?

**Comment by atilaneves** [comment]
@wilzbach I thought of adding a test specific to this, but ended up not doing so because changing the way unit tests are named would at least make some other tests break (since they unfortunately depend on the exact name of unit tests). It's also unlikely to change back so I decided it wasn't worth the effort.

**Comment by wilzbach** [comment]
Ok as 2.078 is about to be released in two days, I think this is too potentially disturbing for a last minute merge and not a regression with absolute priority. Hence, I changed the target branch to master.

@atilaneves I think there is only one nit missing here

**Comment by atilaneves** [comment]
@wilzbach @JinShil I think that should be it. I also rebased.

File: src/dmd/dmodule.d
Comment (JinShil) [inline src/dmd/dmodule.d]: Needs a DDoc `Params`.  Sorry I didn't mention it earlier.

File: src/dmd/func.d
Comment (JinShil) [inline src/dmd/func.d]: Needs a DDoc `Params`.

**Review by JinShil — CHANGES_REQUESTED** [review]

**Comment by atilaneves** [comment]
@JinShil Added the DDoc.

**Review by JinShil — APPROVED** [review]
LGTM though I'd like to see additional support before merging.

File: src/dmd/dmodule.d
Comment (WalterBright) [inline src/dmd/dmodule.d]: module's

File: src/dmd/dmodule.d
Comment (WalterBright) [inline src/dmd/dmodule.d]: `.toString()` should work here, and is much more efficient

**Review by WalterBright — APPROVED** [review]
Approved provided my suggestions are addressed.

File: src/dmd/dmodule.d
Comment (atilaneves) [reply src/dmd/dmodule.d]: Oops. Fixed.

File: src/dmd/dmodule.d
Comment (atilaneves) [reply src/dmd/dmodule.d]: `prependstring` takes a `const(char)*`. I _could_ call `toString` and pass its pointer, but I don't see why nor do I understand why `toString` would be more efficient since the implementations of `toString` and `toChars` are nearly identical:

```d
    override const(char)* toChars() const pure
    {
        return string;
    }

    extern (D) final const(char)[] toString() const pure
    {
        return string[0 .. len];
    }
```

**Comment by atilaneves** [comment]
Can I change the target branch back to stable now that 2.078.0 has been released?

**Comment by wilzbach** [comment]
> Can I change the target branch back to stable now that 2.078.0 has been released?

Sure. Done.

**Pull request merged**
